Alex and Happy Part 2/article.md

```diff
@@ -0,0 +1,1333 @@
+# Parsing with Haskell (Part 2): Parsing with Happy
+
+<!--
+* [X] refer to LR(1) article
+* [X] link to its user guide
+-->
+
+This is the second and final part of the _Parsing with Haskell_ series.
+Haven't read the first part yet? You can find it [here](https://serokell.io/blog/lexing-with-alex).
+
+As with Alex, our goal here is to provide enough information on Happy to be productive and do valuable things with it, hence we won't visit every single feature of Happy.
+Please refer to the [Happy User Guide](https://monlih.github.io/happy-docs/).
+
+This tutorial was written using GHC version 9.0.2, Stack resolver LTS-19.8, Alex version 3.2.7.1, and Happy version 1.20.0.
+
+## Our grammar: MiniML
+
+We've discussed the grammar in the previous article, but let's reintroduce it here.
+MiniML (read as "minimal") will be a small grammar based on ML.
+By the end of this tutorial, you will be able to parse programs such as the following:
+
+```sml
+let the_answer : int =
+  let a = 20 in
+  let b = 1 in
+  let c = 2 in
+  a * c + b * c
+
+let main (unit : ()) : () =
+  print ("The answer is: " + the_answer)
+```
+
+Some of the features that this tutorial will support are:
+* Variables, constants, function declarations, function applications, local bindings, and operators.
+* Some simple types, as well as arrow types (functions).
+* Conditional expressions: if-then-else and if-then.
+* Block comments (added in part 1).
+* The reader will be invited in the proposed exercises to extend the language with patterns, list access, and pattern matching. The exercises in the first part also introduced rational numbers and line comments.
+
+## Comparison with parser combinators
+
+<!--
+* [X] refer to parser combinators article
+* [X] talk about the pros and cons
+-->
+
+In a previous tutorial, we discussed implementing and using [Parser Combinators in Haskell](https://serokell.io/blog/parser-combinators-in-haskell).
```

Comment:
If you do a comparison with Megaparsec et al, (Happy vs. Megaparsec would make our SEO person very happy), I would pull this lower. 

E.g. first you can explain what these libraries are theoretically based on, then explain the differences in the power of those constructs.

Reply:
Renamed it. But I also don't understand your point, what sorts of explanations are you looking for? Like _why_ is LR more powerful than LL? And about what these libraries are based on, I explained a bit more about LR in "How it works", do you want me to say more about how parser combinators and Happy work?